### PR TITLE
Proxy Webpack dev server through to an API server

### PR DESCRIPTION
**Pull request by wchargin**
Summary:
This way, our frontend can talk to a backend that can read from the
filesystem (among other things).

Paired with @decentralion.

Test Plan:
```
$ yarn backend
$ SOURCECRED_DIRECTORY=/tmp/srccrd yarn start
$ # verify that the browser looks good
$ mkdir /tmp/srccrd
$ echo hello >/tmp/srccrd/world
$ curl localhost:3000/api/v1/data/world
hello
$ curl localhost:4000/api/v1/data/world
hello
```

wchargin-branch: webpack-proxy

### CODE REVIEW

File: scripts/start.js [hunk 1]
```diff
@@ -62,48 +70,67 @@ if (process.env.HOST) {
...
+    if (isInteractive) {
+      clearConsole();
+    }
+    console.log(chalk.cyan("Starting the development server...\n"));
+    openBrowser(urls.localUrlForBrowser);
+  });
+
+  ["SIGINT", "SIGTERM"].forEach(function(sig) {
+    process.on(sig, function() {
+      devServer.close();
```
Comment (decentralion) [inline scripts/start.js]: Can we also close the api server

File: scripts/start.js [hunk 2]
```diff
@@ -62,48 +70,67 @@ if (process.env.HOST) {
...
     });
-  })
-  .catch((err) => {
-    if (err && err.message) {
-      console.log(err.message);
+  });
+  console.log(
+    chalk.green(`Server listening on port ${apiServer.address().port}.`)
+  );
+  console.log();
```
Comment (decentralion) [inline scripts/start.js]: ?

File: src/app/apiApp.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// @flow
+
+import express from "express";
+
+export default function apiApp(sourcecredDirectory: string) {
```
Comment (decentralion) [inline src/app/apiApp.js]: can we either:
- remove `node bin/sourcecred.js start`
- or refactor it to depend on this code?

File: scripts/start.js (hunk 2 above)
Comment (wchargin) [reply scripts/start.js]: It prints an empty line, to give some space.

File: src/app/apiApp.js (hunk 3 above)
Comment (wchargin) [reply src/app/apiApp.js]: The new behavior of `node ./bin/sourcecred.js start` will be to start a prod server. I'll implement that presently.

File: scripts/start.js (hunk 1 above)
Comment (wchargin) [reply scripts/start.js]: sure

**Pull request merged**